### PR TITLE
Update go and crossplane-runtime to the latest

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

I had to make a change in Makefile for it to delete only the first line since they don't add empty line in the beginning of YAMLs anymore.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

`make reviewable`

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: cluster/crds/pkg.crossplane.io_controllerconfigs.yaml [hunk 1]
```diff
@@ -1219,6 +1230,17 @@ spec:
                         description: GMSACredentialSpecName is the name of the GMSA
                           credential spec to use.
                         type: string
+                      hostProcess:
```
Comment (hasheddan) [inline cluster/crds/pkg.crossplane.io_controllerconfigs.yaml:1233]: This is a new field so we'll either need to support translating it or craft our own struct with it omitted.

File: go.mod [hunk 2]
```diff
@@ -1,27 +1,114 @@
...
+	k8s.io/apimachinery v0.23.0
+	k8s.io/client-go v0.23.0
+	k8s.io/code-generator v0.23.0
+	k8s.io/utils v0.0.0-20210930125809-cb0fa318a74b
+	sigs.k8s.io/controller-runtime v0.11.0
+	sigs.k8s.io/controller-tools v0.8.0
+	sigs.k8s.io/yaml v1.3.0
+)
+
+require (
```
Comment (negz) [inline go.mod:27]: Is this a Go 1.17 thing?

Comment (muvaf) [reply go.mod:27]: Yes, see `Pruned module graphs in go 1.17 modules` section in https://go.dev/doc/go1.17

File: cluster/crds/pkg.crossplane.io_controllerconfigs.yaml (hunk 1 above)
Comment (muvaf) [reply cluster/crds/pkg.crossplane.io_controllerconfigs.yaml:1233]: It's part of pod security context and that field seems to be set as is [here](https://github.com/crossplane/crossplane/blob/8c48d4c/internal/controller/pkg/revision/deployment.go#L151). So, I think it's handled?

Comment (hasheddan) [reply cluster/crds/pkg.crossplane.io_controllerconfigs.yaml:1233]: Ah good call -- thanks!

**Review by hasheddan — APPROVED** [review]

**Pull request merged**